src/cljs/rems/table.cljs

```diff
@@ -94,20 +140,38 @@
                               (s/optional-key :sort-column) s/Keyword
                               (s/optional-key :sort-order) (s/enum :asc :desc)
                               (s/optional-key :set-sorting) Applicable}
-   (s/optional-key :filtering) {(s/optional-key :filters) {s/Keyword s/Str}
+   (s/optional-key :filtering) {(s/optional-key :filters) s/Str
                                 (s/optional-key :show-filters) s/Bool
                                 (s/optional-key :set-filtering) Applicable}
    :id-function Applicable
    :items [s/Any]
    (s/optional-key :class) s/Str})
 
+(defn- filter-view [filtering]
+  (let [{:keys [filters set-filtering]} filtering]
+    [:div
+      [:input
+       {:type "text"
+        :name "table-search"
+        :value filters
+        :placeholder ""
+        :on-input (fn [event]
+                    (set-filtering (assoc filtering :filters (-> event .-target .-value))))}]
+      (when (seq filters)
+        [:div.reset-button.icon-link.fa.fa-backspace
```

Comment:
Like discussed I would just use a regular button here because now we have space. On the other hand Google does not have a specific clear button so maybe it's not required at all.

Reply:
The state should be reset when closing the search with the search button.

Reply:
Button removed. Disagree about the reset; it's useful to quickly toggle filters on and off.